Highlight previewed items

mehran-naghizadeh opened the pull request:
This PR seeks to solve design problem #36 -> Show last previewed result

mehran-naghizadeh committed:
Create necessary classes

The 'selected' class will be assigned to the last previewd and the 'selected-old' calss will be assigned to other previewd results.

mehran-naghizadeh committed:
Update classList when item is previewed

mehran-naghizadeh committed:
Keep track of what has been previewed

mehran-naghizadeh committed:
Logic for updating the classes

mehran-naghizadeh committed:
Upgrade ember-bootstrap addon's version

mehran-naghizadeh committed:
Get rid of changes made in the last 5 commits

The last five commits were an attempt to solve issue #38 (https://github.com/ipfs-search/ipfs-search-frontend/pull/38). We needed to highlight the last previewed item. However, the implementation had tried to go furthur. It would highlight older previewed items in a different color as well. The problem with this implementation is that it basedd on sharing component instance references. This is seen in the openModal action of the searchhit-card component. Here, this (a reference to the current component) is sent to the calling component (in this case, search-results). THIS IS DANGEROUSLY WRONG.

mehran-naghizadeh committed:
Add  property to search-result

This property will keep track of the last selected/visited search result. It will be set to that result's hash

mehran-naghizadeh committed:
Add  action to search-results

This will update the current selected result record.

mehran-naghizadeh committed:
Inject  action to search-hit-card

This will update the current selected result record when the card is clicked.

mehran-naghizadeh committed:
Add a new `selected` propety and bind style to it

mehran-naghizadeh committed:
Bind  to

This will update the  value (from search-hit-card) to the  value (from searchresult) which contains the hash of last visited result. This will in turn update the style of search-hit-card.

mehran-naghizadeh committed:
Replace the redundant setAsSelectedHit with a mut

mehran-naghizadeh committed:
Rename onclick to onOpenDetail

madnificent commented:
Hi Mehran.  This is starting to look nice.  Summary of our chat.

# onclick callback

The onclick callback should be renamed to something descriptive of the effect, rather than the cause.  We landed on `onOpenDetail`.

# selectedHit name

The `selectedHit` name implies it will contain a hit, rather than the hash.  We'll rename it to `selectedHash` instead.

# setAsSelectedHit action

We can remove this and alter the template to become `{{search-hit-card hit=hit onOpenDetail=(mut selectedHash hit.hash) selected=(eq hit.hash selectedHit)}}`

# check action to exist

We execute the `onOpenDetail` callback argument.  We should check if it exists before calling it.  Something like `if (this.onOpenDetail) this.onOpenDetail();` should be fine.  You can go for `if (typeof this.onOpenDetail === "function" ) { this.onOpenDetail(); }` if you want to be more defensive yet slightly more verbose.

mehran-naghizadeh committed:
Check onOpenDetails validity before calling it

onOpenDetails is a callback argument. So, before calling it, we need to make sure it is valid.

mehran-naghizadeh committed:
Change hover color & last selected item's bg color

madnificent committed:
Merge branch 'v2' into feature/highlight-last-selected
